### DISCUSSION TITLE
Parse Information from CLI to ClearML User Properties

### DISCUSSION THREAD

**Discussion by Sodek93** [ultralytics/yolov5#9702] [Q&A] [answered]
I am logging my YOLOv5 train.py runs with ClearML. 

I use following Code to access CLI to start YOLO:

```
for Resolution in Resolutions:
      for Netsize in Netzsizes:
      string="python yolov5/train.py --img "+Resolution+" --batch 16 --epochs 50 --data dataset.yaml --weights yolov5"+Netsize+".pt --cache"
                  print(string)
                  os.system(str(string))
```

Is it possible to parse some other variables to ClearML? I change dataset.yaml with every iteration too and i would like to track the Pictures i actually use i the current run (i have real and synthetic data and i switch amount every iteration).

Example :
python yolov5/train.py --img 640 --batch 16 --epochs 50 --data dataset.yaml --weights yolov5s.pt **--custom-prob PictureMix5**

I know that a lot of information is already parsed by default (like weights and a lot of others) but i am missing some and can't find a solution for CLI calls.

**Comment by glenn-jocher** [comment]
@thepycoder could you take a look at this question? Thanks!

**Comment by thepycoder** [comment, marked as answer]
Hi @Sodek93 

I don't know if I fully understand your question. As I see it, there are three:

First of all, ClearML will auto-capture command line arguments, so if you add an argument like `--custom-prob PictureMix5`, it should just be picked up! If not, we have a bug on our end on our hands :) Can you check that this argument is present in the configuration section? This should also capture different resolutions and netsizes.


Secondly, changing the dataset yaml every iteration might be a problem, but if its your own code that does that, you can use our logging functions to log the different dataset files as they come in. Check [here](https://clear.ml/docs/latest/docs/clearml_sdk/task_sdk#configuration-objects) for more info.

Third: you want to add custom images. That can be done of course :D Check `report_media` [here](https://clear.ml/docs/latest/docs/guides/reporting/media_reporting) and [here](https://clear.ml/docs/latest/docs/references/sdk/logger#report_media).

So the answer boils down to: yes, most likely you'll be able to using manual logging statements you put in yourself. To do that, check [this file](https://github.com/ultralytics/yolov5/blob/master/utils/loggers/__init__.py). It has a bunch of logging functions that get called from `train.py`. You can either use one of the existing ones or make a new function there and call it in train.py yourself with the arguments you want. 

Most functionality in the reporting file is directed towards the utils file of each respective logger. Ours can be found [here](https://github.com/ultralytics/yolov5/blob/master/utils/loggers/clearml/clearml_utils.py). So it's good practice to put all the actual work (and manual logging statements) there. Those then get called by the logging init, which in turn gets called by `train.py` function by function.

**Reply by Sodek93** [reply]
Thank you for the Answer! Got it working with custom arguments :-)
